src/types.ts

```diff
@@ -34,6 +36,11 @@ export type UmzugOptions<Ctx extends {} = Record<string, unknown>> = {
 		 * in the same folder as the last existing migration. The value here can be overriden by passing `folder` when calling `create`.
 		 */
 		folder?: string;
+		/**
+		 * A function for generating custom prefixes for migration files when using `create`. If this is not specified the default date formats will
+		 * be used ("1970.01.01T00.00.00" for TIMESTAMP, "1970.01.01" for DATE and "" for NONE)
+		 */
+		prefix?: (prefix: MigrationPrefix) => string;
```

Comment:
```suggestion
		prefix?: (params: {name: string}) => string;
```

I like this approach but we can go further. The `MigrationPrefix` type isn't really useful anymore. Let's deprecate it and just pass the name in and let the user define a function that does whatever they need. The default can still be the dot-separated date format that's currently used.